### PR TITLE
provide per particle msd in MSD method

### PR DESCRIPTION
**Pull request by zhou-pj**
## Description
provide user access to per particle based msd in MSD method

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: freud/msd.pyx
Comment (bdice) [inline freud/msd.pyx]: You should check how the docs are rendering this and consider the following suggestion. I don't think you can have a line break in the first section with the math shape and ndarray type. However, if you remove the line break, then flake8 will yell at you unless you add the appropriate `noqa`. This construction ought to work:
```suggestion
        """:math:`\\left(N_{frames}, N_{particles} \\right)` :class:`numpy.ndarray`:
        The per particle based mean squared displacement.
        """  # noqa: E501
```

**Review by bdice — CHANGES_REQUESTED** [review]
I'd like to get a second review from @vyasr and will add him as a reviewer. This needs tests, but otherwise I think it's close to done. Consider adding tests where you have one particle move and another particle that is stationary, or something like that. Also update changelog/credits.

File: freud/msd.pyx
Comment (zhou-pj) [reply freud/msd.pyx]: I am having trouble make the doc and got a couple no module found error, this is the last one(and it doesn't go away when i try install sphinxcontrib): `Could not import extension sphinxcontrib.bibtex (exception: No module named 'sphinxcontrib.bibtex')`. but `make html` works normal on my signac repo. Any idea how I can fix it?

**Comment by zhou-pj** [comment]
there is a existing test on msd accuracy so I just added the comparison on a per particle base as well in that test.

File: freud/msd.pyx
Comment (bdice) [reply freud/msd.pyx]: @zhou-pj You need to install the sphinx extension: `pip install sphinxcontrib-bibtex`. We should probably add this as a "development requirement" but we don't have a `requirements-dev.txt` or similar file right now. Separate issue created: #598.

Comment (zhou-pj) [reply freud/msd.pyx]: thanks! I still have trouble checking the html doc locally though. No matter what I change in the docstring, the html would not change accordingly. Is there another docstring place it is pulling from? Other than that I think the rest is good.

Comment (vyasr) [reply freud/msd.pyx]: Sphinx works by importing the Python module that you're working with and then inspecting the imported object to get docstrings. For pure Python packages (e.g. signac), this means that even if you update the source you have to make sure your PYTHONPATH is pointing to the development version. For freud, this is a little more annoying because pyx files cannot be imported directly, they must be compiled into shared libraries before Python can import them. I'm assuming the reason your documentation is not changing is because you actually need to rerun setup.py to rebuild freud to make the new docstrings accessible to Sphinx. Usually, to avoid continually installing freud into the user site directory I get around this by running `python setup.py build_ext --inplace`. This command will compile the Cython into C++, and compile the resulting C++, but it won't copy freud into your global site-packages folder. This way, you won't be able to import this version of freud from other directories, but you will be able to import it from the root of the repository (`import freud` will see the nested freud folder containing the Cython source files, and all the compiled modules will be placed inside that folder so the import will work). You can then add this folder to your PYTHONPATH. This is very similar to how running HOOMD out of the build directory works.

Comment (zhou-pj) [reply freud/msd.pyx]: I checked the doc it looks good now

**Review by bdice — APPROVED** [review]
Looks good.

**Review by vyasr — APPROVED** [review]
LGTM

**Pull request merged**
